version: '3.8'
services:
  db:
    image: postgres:16.0
    environment:
      POSTGRES_USER: ${DB_USER}
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_DB: ${DB_NAME}
    volumes:
      - db-data:/var/lib/postgresql/data
    ports:
      - "5432:5432"
    env_file:
      - .env
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DB_USER} -d ${DB_NAME}"]
      interval: 10s
      timeout: 5s
      retries: 5

  web:
    build:
      context: .
      dockerfile: dockerization/Dockerfile
    ports:
      - "8080:${PROJECT_PORT}"
    volumes:
      - "./src:/app"
      - "./.env:/app/.env"
    env_file:
      - .env
    command: python3 main.py
    depends_on:
      db:
        condition: service_healthy

volumes:
  db-data:
